src/link/Wasm.zig

```diff
@@ -2202,13 +2224,213 @@ pub fn flush(wasm: *Wasm, comp: *Compilation, prog_node: *std.Progress.Node) lin
         }
         return;
     }
+
     if (build_options.have_llvm and wasm.base.options.use_lld) {
         return wasm.linkWithLLD(comp, prog_node);
+    } else if (build_options.have_llvm and wasm.base.options.use_llvm and !wasm.base.options.use_lld) {
+        return wasm.linkWithZld(comp, prog_node);
     } else {
         return wasm.flushModule(comp, prog_node);
     }
 }
 
+/// Uses the in-house linker to link one or multiple object -and archive files into a WebAssembly binary.
+fn linkWithZld(wasm: *Wasm, comp: *Compilation, prog_node: *std.Progress.Node) link.File.FlushError!void {
```

Comment:
In the future, I think we might want to consider putting this logic (and anything related to traditional link path) into a separate module such as `link/Wasm/zld.zig` or similar to unclutter the code space a little. What do you think?

Reply:
I thought about it, but currently, it doesn't warrant such a change. The logic between the two paths isn't very dissimilar, unlike the MachO linker.